src/plugins/ucc_discovery_plugin.cpp

```diff
@@ -133,24 +134,50 @@ void UccDiscoveryPlugin::_validate_ucc_candidates(const UccCandidates& ucc_candi
 
     const auto& soft_key_constraints = table->soft_key_constraints();
 
-    // Skip already discovered UCCs.
-    if (std::any_of(soft_key_constraints.cbegin(), soft_key_constraints.cend(),
-                    [&column_id](const auto& key_constraint) {
-                      const auto& columns = key_constraint.columns();
-                      return columns.size() == 1 && *columns.cbegin() == column_id;
-                    })) {
-      message << " [skipped (already known) in " << candidate_timer.lap_formatted() << "]";
+    // Check if a primary key constraint already exists for the column. If it does, we can skip the candidate.
+    const auto existing_pk = soft_key_constraints.find(TableKeyConstraint{{column_id}, KeyConstraintType::PRIMARY_KEY});
```

Comment:
We do not expect a PK to be schema-given at the moment. Also, we miss on existing, minimal UCCs when searching for exact matches (e.g., UCC `a, b` can be skipped if `a` is already known). Thus, I guess it would be better to iterate the table constraints and look for containment of the column IDs.